### 1) Вероятность того, что стрелок попадет в мишень, выстрелив один раз, равна 0.8. Стрелок выстрелил 100 раз. Найдите вероятность того, что стрелок попадет в цель ровно 85 раз.

Для решения данной задачи воспользуемся формулой *Бернулли*:

In [7]:
from math import factorial, exp
def bernulli(n, k, p):
    comb=factorial(n)/(factorial(k)*factorial(n-k))
    return comb*(p**k)*(1-p)**(n-k)

In [6]:
print(f'P = {bernulli(100,85,0.8): .4f}')

P =  0.0481


###### Ответ:
**Вероятность того, что стрелок попадет в цель ровно 85 раз: P =  0.0481 = 4.81%**

### 2) Вероятность того, что лампочка перегорит в течение первого дня эксплуатации, равна 0.0004. В жилом комплексе после ремонта в один день включили 5000 новых лампочек.
#### а)Какова вероятность, что ни одна из них не перегорит в первый день? 
#### б) Какова вероятность, что перегорят ровно две?

Поскольку в задаче вероятность наступления события мала, а количество испытаний велико, для решения задачи воспользуемся формулой Пуассона:

In [3]:
def poisson_distr(m,p,n):
    lamb=p*n
    return exp(-lamb)*(lamb**m)/factorial(m)

a) По условию задания: p = 0.0004n = 5000m = 0, тогда:

In [16]:
print(f'P = {poisson_distr(0,0.0004,5000): .4f}')

P =  0.1353


##### Ответ:
**Вероятность того, что ни одна из ламп не перегорит в первый день: P = 0.1353 = 13.53%**

б) По условию задания: p = 0.0004n = 5000m = 2, тогда:

In [5]:
print(f'P = {poisson_distr(2,0.0004,5000): .4f}')

P =  0.2707


##### Ответ:
**Вероятность того, что 2 лампы перегорят в первый день: P = 0.2707 = 27.07%**

### 3) Монету подбросили 144 раза. Какова вероятность, что орел выпадет ровно 70 раз?

Воспользуемся формулой Бернулли, в нашем случае: _n = 144k = 70p = 0.5_

Вычислим:

In [8]:
print(f'P = {bernulli(144,70,0.5): .4f}')

P =  0.0628


##### Ответ:
**Вероятность того, что при 144-х кратном подбрасывании монетки орёл выпадет ровно 70 раз: P = 0.0628 = 6.28%**

### 4) В первом ящике находится 10 мячей, из которых 7 - белые. Во втором ящике - 11 мячей, из которых 9 белых. Из каждого ящика вытаскивают случайным образом по два мяча.
#### а) Какова вероятность того, что все мячи белые? 
#### б) Какова вероятность того, что ровно два мяча белые? 
#### в) Какова вероятность того, что хотя бы один мяч белый?

**_Для решения данной задачи подсчитаем общее число элементарных исходов при вытаскивании мячей._**

   **_Напишем функцию, для подсчёта количества комбинаций:_**

In [9]:
def combinations(n, k):
    return int(factorial(n) / (factorial(k) * factorial(n - k)))

Вычислим n1 и n2:

In [11]:
n1=combinations(10,2)
n2=combinations(11,2)
print(f'Общее число элементарных исходов для первой корзины: n1 = {n1}\n'
f'Общее число элементарных исходов для второй корзины: n2 = {n2}')

Общее число элементарных исходов для первой корзины: n1 = 45
Общее число элементарных исходов для второй корзины: n2 = 55


**_a) Для выполнения этого условия необходимо одновременное наступление двух событий:_**

**_событие A-из первой корзины извлекли 2 белых шара,_**

**_событие B-из второй корзины извлекли 2 белых шара._**

Вычислим m1 и m2:

In [17]:
m1=combinations(7,2)
m2=combinations(9,2)
print(f'Число благоприятных элементарных исходов для события А: m1 = {combinations(7,2)}\n'
      f'Число благоприятных элементарных исходов для события В: m2 = {combinations(9,2)}')

Число благоприятных элементарных исходов для события А: m1 = 21
Число благоприятных элементарных исходов для события В: m2 = 36


Для событий A и B вероятности их наступления будут расчитываться как
_P(A)=m1/n1_  и _P(A)=m2/n2_, а вероятность их одновременного наступления как
_P(A+B)=P(A)*P(B)_ 

Расчитаем:

In [14]:
print(f'Вероятность наступления события А, Р(А) = {m1/n1: .4f}\n'
      f'Вероятность наступления события В, Р(В) = {m2/n2: .4f}\n'
      f'Вероятность одновременного наступления событий А и В, Р(А+В) = {(m1/n1)*(m2/n2): .4f}'
     )

Вероятность наступления события А, Р(А) =  0.4667
Вероятность наступления события В, Р(В) =  0.6545
Вероятность одновременного наступления событий А и В, Р(А+В) =  0.3055


**Вариант решения через условную вероятность:**

In [15]:
P=(7/10)*(6/9)*(9/11)*(8/10)
print(f'Вероятность одновременного наступления событий А и В, Р(А+В) = {P: .4f}')

Вероятность одновременного наступления событий А и В, Р(А+В) =  0.3055


##### Ответ:
**Вероятность того, что все извлечённые мячи белые: P(A+B) = 0.3055 = 30.55%**

**_б) Этому варианту соответствует несколько комбинаций:_**

1. События A-из первой корзины извлекли 2 белых шара и B-из второй корзины извлекли 2 не белых шара, или
2. События C-из первой корзины извлекли 1 белый и 1 не белый шар и D-из второй корзины извлекли 1 белый и 1 не белый шар, или
3. События E-из первой корзины извлекли 2 не белых шара и F-из второй корзины извлекли 2 белых шара

А вероятность того, что сложится одна из этих комбинаций, будет суммой вероятностей этих комбинаций. Т.е.:

**P = P(A) * P(B) + P(C) * P(D) + P(E) * P(F)**

Число благоприятных элементарных исходов для каждого события в комбинации:

In [18]:
m_A=combinations(7,2)
m_B=combinations(2,2)
m_C=combinations(7,1)*combinations(3,1)
m_D=combinations(9,1)*combinations(2,1)
m_E=combinations(3,2)
m_F=combinations(9,2)
print(f'Число благоприятных элементарных исходов для события А: m_A = {m_A}\n'
      f'Число благоприятных элементарных исходов для события B: m_B = {m_B}\n'
      f'Число благоприятных элементарных исходов для события C: m_C = {m_C}\n'
      f'Число благоприятных элементарных исходов для события D: m_D = {m_D}\n'
      f'Число благоприятных элементарных исходов для события E: m_E = {m_E}\n'
      f'Число благоприятных элементарных исходов для события F: m_F = {m_F}'  
     )

Число благоприятных элементарных исходов для события А: m_A = 21
Число благоприятных элементарных исходов для события B: m_B = 1
Число благоприятных элементарных исходов для события C: m_C = 21
Число благоприятных элементарных исходов для события D: m_D = 18
Число благоприятных элементарных исходов для события E: m_E = 3
Число благоприятных элементарных исходов для события F: m_F = 36


In [19]:
P=(m_A/n1)*(m_B/n2)+(m_C/n1)*(m_D/n2)+(m_E/n1)*(m_F/n2)
print(f'Вероятность того, что будут вытащены ровно 2 белых мяча Р = {P: .4f}')

Вероятность того, что будут вытащены ровно 2 белых мяча Р =  0.2048


**Вариант решения через условную вероятность:**

In [20]:
P1=((7/10)*(6/9)*(2/11)*(1/10))+4*((7/10)*(3/9)*(9/11)*(2/10))+((3/10)*(2/9)*(9/11)*(8/10))
print(f'Вероятность того, что будут вытащены ровно 2 белых мяча Р = {P1: .4f}')

Вероятность того, что будут вытащены ровно 2 белых мяча Р =  0.2048


##### Ответ:
**Вероятность того, что будут вытащены ровно 2 белых мяча: P = 0.2048 = 20.48%**

**_в) В данном случае удобнее рассмотреть событие Ā -"не было извлечено ни одного белого мяча", которое наряду с искомым событием A -"был извлечен хотя бы 1 белый мяч", составляют полную группу событий, по скольку в результате испытаний может произойти только либо событие A, либо событие Ā, и поэтому:_**

**P(A)=1−P(Ā)**

Для наступления события Ā необходимо чтобы из первой и второй корзин не было вынуто ни одного белого мяча, число таких благоприятных исходов для первой корзины m1=e_3^2, а для второй e_2^2

Вычислим m1 и m2:

In [21]:
m1=combinations(3,2)
m2=combinations(2,2)
print(f'Число благоприятных элементарных исходов для первой корзины: m1 = {combinations(3,2)}\n'
      f'Число благоприятных элементарных исходов для второй корзины: m2 = {combinations(2,2)}'
     )

Число благоприятных элементарных исходов для первой корзины: m1 = 3
Число благоприятных элементарных исходов для второй корзины: m2 = 1


** Тогда P(Ā)=m1/n1+  m2/n2, вычислим P(Ā) и P(A):**

In [23]:
print(f'Вероятность того, что не будет извлечено ни одного белого мяча Р(А_обр) = {(m1/n1)*(m2/n2): .4f}\n'
      f'Вероятность того, что будет извлечен хотя бы один белый мяч Р(А) = 1 - Р(А_обр) = {1-(m1/n1)*(m2/n2): .4f}'
     )

Вероятность того, что не будет извлечено ни одного белого мяча Р(А_обр) =  0.0012
Вероятность того, что будет извлечен хотя бы один белый мяч Р(А) = 1 - Р(А_обр) =  0.9988


**Вариант решения через условную вероятность:**

In [24]:
P=1-(3/10)*(2/9)*(2/11)*(1/10)
print(f'Вероятность того, что будет извлечен хотя бы один белый мяч Р(А) = {P: .4f}')

Вероятность того, что будет извлечен хотя бы один белый мяч Р(А) =  0.9988


##### Ответ: 
**Вероятность того, что будет извлечен хотя бы один белый мяч P = 0.9988 = 99.88%**